.github/workflows/lib-codeql.yaml

```diff
@@ -25,11 +25,11 @@ jobs:
         check-latest: true
 
     - name: Initialize CodeQL
-      uses: github/codeql-action/init@2c779ab0d087cd7fe7b826087247c2c81f27bfa6 # v3
+      uses: github/codeql-action/init@4dd16135b69a43b6c8efb853346f8437d92d3c93 # v3
```

Comment:
@mythi I'd welcome the `v3` here as well to avoid changing it all the time, what do you think?

Reply:
I'm proposing we move away from tags to sha:s for [added security](https://docs.github.com/en/actions/security-for-github-actions/security-guides/security-hardening-for-github-actions#using-third-party-actions. "all the time" is tunable in Dependabot settings: it's no weekly but it can be monthly of course as well.

Reply:
The question is whether we are checking the changes in those GH actions before updating them. TBH I have not reviewed the history between `2c77` and `4dd1` as I do trust `codeql-action` organization.

We can either be paranoiac but then we need strict guidelines to update the dependencies (basically audit or at least review all the to-be-updated actions) but without such guidelines I don't see much sense in the additional overhead.

In the real-life I'd go with something in the middle meaning we can identify organizations we trust and use tags there (like `action/checkout`) and start using hashes for those we don't trust and add a comment reminding us to review that action code (or at least history) before updating and mention it's for security reasons. Would that work for you?

Reply:
If we accommodate the shas with simple comment reminding reviewers to check the GH action log then I'm fine with using shas. Although I'd only do that where needed as every change needs to be reviewed by 2 people, CI needs to be performed (even though it won't test the change as we run on `pull_request_target`) so yes, it takes time.

Anyway I'm only speaking from my point of view and I'm not maintaining any self-provisioned runners so I think this question should be raised among those who take care of them.

Reply:
@ldoktor one clarification to this topic overall: I feel there might be a misunderstanding what these SHAs are. They are always pointing to some released version. We don't get updates every time something gets committed to the action repo (which I think might be the reason for your comment: "changing it all the time").

Reply:
> Yeah I am not really convinced by the benefits of using hashes.

I've moved to use them in a few places. Not because of being paranoid but mainly because it's considered a good security practice. That practice also checked by tools like OpenSSF Scorecard so if our repos are benchmarked using that common tooling, it shows we are "compliant".

After all, the automation takes care of the updates and we get them on a regular basis (like this PR). If the weekly cadence is too high, we can make it monthly.

Reply:
One thing I haven't found in docs (maybe I'm blind) is whether by choosing hashes dependabot will offer the latest main, latest tagged or latest tagged of the version that was previously tagged sha? Or if it uses the `comment` next to it as "stream" guidance?

Als I'd really like to know what is expected of us during the review. Should it be just `yeah, dependabot, merge this` or `let's check the changelog` or `review the commit log` or `look at the git diff of that action and look for potential threats`. If we decide we have different needs for different actions then I'd like the "non-default" review policy to be stated in comment, something like:

```
# Use `git diff` to check changes on this GH action to prevent security issues
uses: github/codeql-action/init@4dd16135b69a43b6c8efb853346f8437d92d3c93 # v3
```

Reply:
> latest tagged or latest tagged of the version that was previously tagged sha?

I don't know what the latter means. My experience is it picks up the sha corresponding to the latest tag. The Actions tab has logs, maybe that helps.

> I'd really like to know what is expected of us during the review

this is separate from the sha conversation. I'm usually reading the changelog added in the cover letter.

Reply:
Thanks, I got the same impression (latest tagged sha).

It's actually important to know what is expected of us to be reviewed. Without that discussion there is no guideline to merge this PR (I know we merged the other PRs already but here we already have some people to discuss it with, which is why I'd like to finish this discussion first).

Reply:
Yes, that's ok. I don't have any other review guideline suggestions to add. Most of coco repos have had dependabot enabled and it's been working well. Here we probably want to enable it for Go dependencies too.